qiskit_dynamics/signals/signals.py

```diff
@@ -569,7 +546,7 @@ def __init__(self, *signals, name: Optional[str] = None):
             elif isinstance(sig, Signal):
                 components.append(sig)
             elif isinstance(sig, (int, float, complex)) or (
-                isinstance(sig, Array) and sig.ndim == 0
+                hasattr(sig, "__array__") and sig.ndim == 0
```

Comment:
`ArrayLike` cannot be used in `isinstance`. So we have no choice but to use `hasattr`.